### PR TITLE
Add Makefile and ignore for env config

### PR DESCRIPTION
**Pull request by phughk**
Add a Makefile to the project that allows for convenient building of server and client zip files.
Additionally, include the `.env` file to `.gitignore` that is used to configure the Makefile variables for environments.

Now it is possible to run `make install` and find the resulting client and server zips in the target directory.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+TARGET_DIR ?= deploy/
+
+.PHONY: install
+install:
+	# Building project
+	mvn clean install package -DskipTests
+	# Packaging Mage.Client to zip
+	cd Mage.Client && mvn assembly:assembly -DskipTests
+	# Packaging Mage.Server to zip
+	cd Mage.Server && mvn assembly:assembly
```
Comment (phughk) [inline Makefile:16]:
```suggestion
	cd Mage.Server && mvn assembly:assembly -DskipTests
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+TARGET_DIR ?= deploy/
+
+.PHONY: install
+install:
+	# Building project
+	mvn clean install package -DskipTests
+	# Packaging Mage.Client to zip
+	cd Mage.Client && mvn assembly:assembly -DskipTests
+	# Packaging Mage.Server to zip
+	cd Mage.Server && mvn assembly:assembly -DskipTests
```
Comment (JayDi85) [inline Makefile:16]: Tests aren't used in package/assembly phase, so no needs in skip param here

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+include .env
+
+# The target directory is used for setting where the output zip files will end up
```
Comment (JayDi85) [inline Makefile:3]: Just for info:

Real xmage bundle must be a single zip file with both client and server folders, see `./Utils/build-and-package.pl` as build script example.

Current beta server uses another scripts to build and prepare (folders only for updates and additional launcher files for full bundle):
shot_231009_092952
shot_231009_092946

**Review by JayDi85 — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (phughk) [reply Makefile]: Removed skips

File: Makefile (hunk 3 above)
Comment (phughk) [reply Makefile:3]: Added note that the perl script is source of truth and this is just helper

**Pull request merged**